nuget/lib/dependabot/nuget/file_parser/project_file_parser.rb

```diff
@@ -179,21 +192,53 @@ def add_transitive_dependencies_from_project_references(project_file, doc, depen
 
             full_project_path = full_path(project_file, relative_path)
 
-            referenced_file = dependency_files.find { |f| f.name == full_project_path }
-            next unless referenced_file
-
-            dependency_set(project_file: referenced_file).dependencies.each do |dep|
-              dependency = Dependency.new(
-                name: dep.name,
-                version: dep.version,
-                package_manager: dep.package_manager,
-                requirements: []
-              )
-              dependency_set << dependency
+            full_project_paths = expand_wildcards_in_project_reference_path(full_project_path)
+
+            full_project_paths.each do |path|
+              # Check if we've already visited this project file
+              next if visited_project_files.include?(path)
+
+              visited_project_files.add(path)
+              referenced_file = dependency_files.find { |f| f.name == path }
+              next unless referenced_file
+
+              dependency_set(project_file: referenced_file,
+                             visited_project_files: visited_project_files).dependencies.each do |dep|
+                dependency = Dependency.new(
+                  name: dep.name,
+                  version: dep.version,
+                  package_manager: dep.package_manager,
+                  requirements: []
+                )
+                dependency_set << dependency
+              end
             end
           end
         end
 
+        sig { params(full_path: T.untyped).returns(T::Array[T.nilable(String)]) }
+        def expand_wildcards_in_project_reference_path(full_path)
+          return [full_path] unless full_path.include?("*") || full_path.include?("?")
```

Comment:
You can skip this shortcut.  `Dir.glob()` will return just a single file if that's all that was specified.  E.g., calling it with `some/file.txt` will return `['some/file.txt']` and calling it with `some/*.txt` will return `['some/file.txt', 'some/other-file.txt']`.